diff --git a/frontend/src/socket.js b/frontend/src/socket.js
index a942ea6fc..aad13cef4 100644
--- a/frontend/src/socket.js
+++ b/frontend/src/socket.js
@@ -1,5 +1,4 @@
 import { io } from "socket.io-client"
-import { socketio_port } from "../../../../sites/common_site_config.json"
 
 import { getCachedListResource } from "frappe-ui/src/resources/listResource"
 import { getCachedResource } from "frappe-ui/src/resources/resources"
@@ -7,9 +6,7 @@ import { getCachedResource } from "frappe-ui/src/resources/resources"
 export function initSocket() {
 	let host = window.location.hostname
 	let siteName = window.site_name
-	let port = window.location.port ? `:${socketio_port}` : ""
-	let protocol = port ? "http" : "https"
-	let url = `${protocol}://${host}${port}/${siteName}`
+	let url = `https://${host}/${siteName}`
 	let socket = io(url, {
 		withCredentials: true,
 		reconnectionAttempts: 5,
